src/tests/networking_tests.rs

```diff
@@ -641,17 +903,139 @@ pub mod net_tests {
                 s_addr: u32::from_ne_bytes([127, 0, 0, 1]),
             },
             padding: 0,
-        }); //127.0.0.1
+        }); // 127.0.0.1
 
+        let mut retsocket = interface::GenSockaddr::V4(interface::SockaddrV4::default());
         assert_eq!(cage.bind_syscall(sockfd, &socket), 0);
         assert_eq!(cage.getsockname_syscall(sockfd, &mut retsocket), 0);
         assert_eq!(retsocket, socket);
 
-        //checking that we cannot rebind the socket
+        // checking that we cannot rebind the socket
         assert_eq!(cage.bind_syscall(sockfd, &socket), -(Errno::EINVAL as i32)); //already bound so should fail
         assert_eq!(cage.getsockname_syscall(sockfd, &mut retsocket), 0);
         assert_eq!(retsocket, socket);
 
+        assert_eq!(cage.fork_syscall(2), 0); // used for AF_INET thread client
+
+        let barrier = Arc::new(Barrier::new(2));
+        let barrier_clone = barrier.clone();
+
+        let threadclient1 = interface::helper_thread(move || {
+            let cage2 = interface::cagetable_getref(2);
+            assert_eq!(cage2.close_syscall(sockfd), 0);
+
+            barrier_clone.wait();
+            // connect to server
+            assert_eq!(cage2.connect_syscall(clientsockfd1, &socket), 0);
+
+            let mut retsocket = interface::GenSockaddr::V4(interface::SockaddrV4::default());
+            assert_eq!(cage2.getsockname_syscall(clientsockfd1, &mut retsocket), 0);
+            // BUG: when calling connect without binding to any address, an new address will
+            // be automatically assigned to the socket, and its ip address is supposed to be
+            // 127.0.0.1, and port is not supposed to be 0. But we failed this test
+
+            // assert_ne!(retsocket.port(), 0);
+            // assert_eq!(retsocket.addr(), interface::GenIpaddr::V4(interface::V4Addr {
+            //     s_addr: u32::from_ne_bytes([127, 0, 0, 1]),
+            // }));
+
+            cage2.exit_syscall(EXIT_SUCCESS);
+        });
+
+        assert_eq!(cage.listen_syscall(sockfd, 4), 0);
+        let mut sockgarbage = interface::GenSockaddr::V4(interface::SockaddrV4::default());
+        barrier.wait();
+        let fd = cage.accept_syscall(sockfd as i32, &mut sockgarbage);
+        assert!(fd > 0);
```

Comment:
This is a little pedantic, but why > 0?  I think you want fd >= 0, fd !=-1, fd > -1, etc.

Reply:
This is something I copied from other test and did not notice. But I actually feel fd > 0 makes some sense, since 0 is supposed to be occupied by stdin, and we did not do things like dup or close before it, so I guess the fd should not be equal to 0 as well?

Reply:
What about fd 1 and 2 though?  They should be occupied too.  Why is fd 0 special?

Reply:
yeah, I guess I'll remove this check, it is not the focus of this test after all